File: packages/typescript/test/smoke.mjs
import assert from 'node:assert/strict'
import { createRequire } from 'node:module'
import { VendorEvidenceDrift as EsmClient } from '../dist/index.js'

const require = createRequire(import.meta.url)
const { VendorEvidenceDrift: CjsClient } = require('../dist/index.cjs')
let calls = 0
const fakeFetch = async (url, init) => {
  calls += 1
  assert.equal(url, 'https://mock.invalid/health')
  assert.equal(init.method, 'GET')
  return { ok: true, status: 200, text: async () => '{"ok":true,"product":"fixture","version":"1.0.0"}' }
}
const esm = new EsmClient({ apiKey: 'nonsecret-sentinel', baseUrl: 'https://mock.invalid', fetch: fakeFetch })
const cjs = new CjsClient({ apiKey: 'nonsecret-sentinel', baseUrl: 'https://mock.invalid', fetch: () => { throw new Error('unexpected construction-time network') } })
assert.ok(esm)
assert.ok(cjs)
assert.equal(calls, 0)
await esm.health()
assert.equal(calls, 1)
console.log('npm smoke: PASS')

File: packages/python/test/smoke.py
import json
from unittest.mock import patch
from vendor_evidence_drift import VendorEvidenceDrift

calls = []
class Response:
    def __enter__(self): return self
    def __exit__(self, *_args): return False
    def read(self): return json.dumps({"ok": True, "product": "fixture", "version": "1.0.0"}).encode()

def fake_urlopen(request, *args, **kwargs):
    calls.append(request.full_url)
    assert request.full_url == "https://mock.invalid/health"
    return Response()

with patch("urllib.request.urlopen", side_effect=fake_urlopen):
    client = VendorEvidenceDrift("nonsecret-sentinel", base_url="https://mock.invalid")
    assert calls == []
    result = client.health()
    assert result["ok"] is True
    assert calls == ["https://mock.invalid/health"]
print("pypi smoke: PASS")
